sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSQLInsertTestSuite.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive
+
+import org.apache.spark.sql.SQLInsertTestSuite
+import org.apache.spark.sql.hive.test.TestHiveSingleton
+
+class HiveSQLInsertTestSuite extends SQLInsertTestSuite with TestHiveSingleton {
```

Comment:
We need to set ` spark.conf.set("hive.exec.dynamic.partition.mode", "nonstrict")` before executing this suite; otherwise, `test("insert with column list - follow table output order + partitioned table")` will fail. 

The reason why it does not fail because some test cases [running before this suite] do not change the default value of `hive.exec.dynamic.partition.mode` back to `strict`. However, the order of test suite execution is not deterministic.

Reply:
How about something like

```
  protected override def beforeAll(): Unit = {
    spark.conf.set("hive.exec.dynamic.partition.mode", "nonstrict")
    super.beforeAll()
  }

  protected override def afterAll(): Unit = {
    try {
      spark.conf.set("hive.exec.dynamic.partition.mode", "strict")
    } finally {
      super.afterAll()
    }
  }
```

Reply:
Good catch! We can follow `HiveCharVarcharTestSuite` to set/restore this conf.

Reply:
I have submitted a followup